synapse/lib/auth.py

```diff
@@ -264,6 +264,19 @@ async def getRoleByName(self, name):
     def _getRoleIden(self, name):
         return self.roleidenbyname.get(name)
 
+    # TODO convert getUserByName() and getRoleByName()
+    # back from async? These were plumbed to avoid infecting
+    # type norm/repr functions with async...
```

Comment:
Yeah I think we should do that when the deprecated `auth:ctor` config option is removed, no reason for them to be async